WordPress/gutenberg
# Button Block: Use hook based border support

## Description
Part of #28913.

Remove the custom border radius panel from the Button block's `edit()`, and opt that button into using `__experimentalBorder` block supports for border radius instead. Furthermore, enable border radius support for the Button block in `experimental-default-theme.json` in order to preserve the current user-facing behavior.

## How has this been tested?

- Make sure you're on the `trunk` branch, create a new post, and insert a Buttons Block.
- In the Buttons Block, create a button (by typing a label).
- Note the "Border settings" panel in the sidebar. Change the border radius by using the slider, and save the post. Make note of the border radius value that you assigned.
- Switch to this branch, build it, and reload the editor, with the post that you just created still open.
- Verify that the Buttons block loads as before, and that the border radius in the sidebar panel is the one you assigned earlier.
- Preview the post, and verify (e.g. in the DOM inspector) that the border radius is the one you assigned.

## Screenshots

![button-block-borders](https://user-images.githubusercontent.com/96308/112884139-7033b900-90cf-11eb-9162-0fdb0280f805.gif)
